### PR TITLE
ICU-22324 Mavenization, step 2, reorg folders & workflows

### PR DESCRIPTION
**Pull request by mihnita**
##### Checklist

- [x] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-22324
- [x] Required: The PR title must be prefixed with a JIRA Issue number. 
- [x] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [x] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [x] Issue accepted (done by Technical Committee after discussion)
- [x] Tests included, if applicable
- [x] API docs and/or User Guide docs changed or added, if applicable

ALLOW_MULTIPLE_COMMITS=true

### CODE REVIEW

File: .github/workflows/icu_ci.yml [hunk 1]
```diff
@@ -34,7 +40,7 @@ jobs:
           set +o pipefail && make doc 2>&1 | tee doxygen.log && ( ! grep -P 'warning:(?! .* file .?Doxyfile)' doxygen.log )
 
   # ICU4J build and unit test using Ant
```
Comment (markusicu) [inline .github/workflows/icu_ci.yml:42]: s/using Ant/using Maven/

File: .github/workflows/icu_ci.yml [hunk 2]
```diff
@@ -54,12 +60,10 @@ jobs:
       - name: ICU4J
         run: |
           cd icu4j;
-          ant init;
-          ant check;
-          ant localespiCheck
+          mvn install
```
Comment (markusicu) [inline .github/workflows/icu_ci.yml:63]: Can developers normally do something like `mvn check` or is it always `mvn install` now?

**Review by markusicu — DISMISSED** [review]
rslgtm

File: .github/workflows/icu_ci.yml (hunk 2 above)
Comment (echeran) [reply .github/workflows/icu_ci.yml:63]: There is no `mvn check`. You can run:

-  `mvn test` to only run unit tests
-  `mvn integration-test` or `mvn package` to run unit tests & integration tests

We have info on usage already in https://github.com/unicode-org/icu/blob/main/icu4j/maven-build/README.md. Mihai has more docs on his local machine (not visible in a PR yet) that he is preparing for Step 4 that will go into further detail to allow people to understand why/how about Maven, beyond just the usage.

File: .github/workflows/icu_ci.yml (hunk 1 above)
Comment (mihnita) [reply .github/workflows/icu_ci.yml]: Done

**Review by markusicu — DISMISSED** [review]

**Review by echeran — APPROVED** [review]
LGTM

**Review by echeran — APPROVED** [review]

**Review by markusicu — APPROVED** [review]
lgtm tnx!

**Comment by markusicu** [comment]
FYI you should be able to ignore the single-commit failure because I added `ALLOW_MULTIPLE_COMMITS=true` to the PR description.

**Comment by FrankYFTang** [comment]
Looks like all the icu4j-XXXperf test were break after the landing of this PR with message such as
"No benchmark result was found in /home/runner/work/icu/icu/icu4j/perf-tests/perf/results/j_converterperf/s-chinese/UTF-8/TestCharsetDecoderICU/output.txt. Benchmark output was ''"

**Comment by FrankYFTang** [comment]
Bug file under https://unicode-org.atlassian.net/browse/ICU-22477 about the j-perf tests breakage

**Pull request merged**